pkg/arm/resources.go

```diff
@@ -385,8 +385,8 @@ func nsgMaster(cs *api.OpenShiftManagedCluster) *network.SecurityGroup {
 						Description:              to.StringPtr("Allow SSH traffic"),
 						Protocol:                 network.SecurityRuleProtocolTCP,
 						SourcePortRange:          to.StringPtr("*"),
-						DestinationPortRange:     to.StringPtr("22-22"),
-						SourceAddressPrefix:      to.StringPtr("*"),
+						DestinationPortRange:     to.StringPtr(pc.SSHDestPortRange),
+						SourceAddressPrefix:      to.StringPtr(pc.SSHSrcAddrPrefix),
```

Comment:
Bet there'll be multiple ranges in prod.  Can you try using SourceAddressPrefixes ([]string)?